avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -451,11 +452,30 @@ private static Object fromConnectData(
               requireContainer);
 
         case BYTES: {
-          ByteBuffer bytesValue = value instanceof byte[] ? ByteBuffer.wrap((byte[]) value) :
-                                  (ByteBuffer) value;
+          value = value instanceof byte[] ? ByteBuffer.wrap((byte[]) value) :
+            (ByteBuffer) value;
+          if (schema != null && schema.parameters() != null
+              && schema.parameters().containsKey(CONNECT_AVRO_FIXED_SIZE)) {
+            int size = Integer.parseInt(schema.parameters().get(CONNECT_AVRO_FIXED_SIZE));
+            org.apache.avro.Schema fixedSchema = null;
+            if (avroSchema.getType() == org.apache.avro.Schema.Type.UNION) {
+              for (org.apache.avro.Schema memberSchema : avroSchema.getTypes()) {
+                if (memberSchema.getType() == org.apache.avro.Schema.Type.FIXED
+                    && memberSchema.getFixedSize() == size) {
```

Comment:
Here, actually, is the name vs size vs both test.

Reply:
yeah, there is the possible case of two different fixed schemas of the same size but different name in the same union.